lxd/instance_delete.go

```diff
@@ -88,15 +96,29 @@ func instanceDelete(d *Daemon, r *http.Request) response.Response {
 	}
 
 	resources := map[string][]api.URL{}
-	resources["instances"] = []api.URL{*api.NewURL().Path(version.APIVersion, "instances", name)}
+	if deploymentName != "" && deploymentShapeName != "" {
+		resources["instances"] = []api.URL{*api.NewURL().Path(version.APIVersion, "deployments", deploymentName, "shapes", deploymentShapeName, "instances", name)}
+	} else {
+		resources["instances"] = []api.URL{*api.NewURL().Path(version.APIVersion, "instances", name)}
+	}
 
 	if inst.Type() == instancetype.Container {
 		resources["containers"] = resources["instances"]
 	}
 
-	op, err := operations.OperationCreate(s, projectName, operations.OperationClassTask, operationtype.InstanceDelete, resources, nil, rmct, nil, nil, r)
-	if err != nil {
-		return response.InternalError(err)
+	var op *operations.Operation
+	if deploymentName != "" && deploymentShapeName != "" {
```

Comment:
IMHO rather than duplicating the operations.OperationCreate call here, it would be clearer to move that below the if/else, and instead assign differences to variables that can then be passed into a single OperationCreate call.